src/multidecoder/decoders/network.py

```diff
@@ -1076,6 +1076,13 @@ def find_domains(data: bytes) -> list[Node]:
             # If it is, we need to remove the trailing characters that follow as that's not part of the actual domain.
             domain = domain[2:]
             start = start + 2
+        elif preceeding_character == "/":
+            # The match we found might be part of the path of a URL
+            if re.match(
+                rb"(?r)[^:\/]\/([%0-9A-F:]|[a-zA-Z0-9_.-])*" + domain, data, pos=start - 128, endpos=end
+            ) and re.match(domain + rb"([%0-9A-F:]|[a-zA-Z0-9_.-])*(\/)?", data, pos=start, endpos=end + 128):
```

Comment:
```suggestion
            if re.match(rb"(?r)[^:\/]\/([%:a-zA-Z0-9_.-]){,128}", data, endpos=start):
```
- The second regex matches empty and so matches anything after the domain. It's unnecessary.
- We don't need to have the domain in the regex.
- Alternations of character classes is equivalent to a combined character class
- Length to check can be moved into the regex.